pkg/build/generator/generator.go

```diff
@@ -735,3 +736,53 @@ func getImageChangeTriggerForRef(bc *buildapi.BuildConfig, ref *kapi.ObjectRefer
 	}
 	return nil
 }
+
+func generateBuildTriggerInfo(requestBuildCauses []*buildapi.BuildTriggerCauses) (buildReasons []*buildapi.BuildTriggerInfo) {
+
+	for _, reason := range requestBuildCauses {
+		switch reason.TriggerType {
+		// the default behavior for no triggertype is a manual build triggered by
+		// the user whenever the user does start-build.
+		// NOTE: This needs either a new const TriggerType that covers jenkins or
+		// another way to differentiate  between builds triggered by a user
+		// clicking a button in the webui, doing oc new-app/new-build/start-build
+		// and jenkins triggering a new build
+		default:
```

Comment:
@mfojtik  i was looking at switch in the golang docs https://golang.org/doc/effective_go.html#switch , and if you look all the way at the bottom at "Type Switch" default is first, so that sort of stuck with me for some reason, i know in other languages it's at the bottom usually. but will refactor

@rhcarvalho you're right, i just didn't want to go around the API and start defining constants for each trigger type as i please, as i know nobody would like that so i worked with what i had. also i think there should never be an unknown build trigger type